### PR TITLE
fix: data residency

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

- Made data plane Url optional in load call.
- Update unit test cases

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: packages/analytics-js/src/components/core/Analytics.ts [hunk 1]
```diff
@@ -129,14 +130,20 @@ class Analytics implements IAnalytics {
...
+  load(writeKey: string, dataPlaneUrl?: string, loadOptions: Partial<LoadOptions> = {}) {
     if (state.lifecycle.status.value) {
       return;
     }
 
     // Attach global error boundary handler
     this.attachGlobalErrorHandler();
 
+    // dataPlaneUrl is not provided
+    if (typeof dataPlaneUrl === 'object' && dataPlaneUrl !== null) {
```
[the remark this answers is not in the document]
Comment (bardisg) [reply packages/analytics-js/src/components/core/Analytics.ts:142]: isObjectLiteralAndNotNull

File: packages/analytics-js/src/components/core/Analytics.ts [hunk 2]
```diff
@@ -129,14 +130,20 @@ class Analytics implements IAnalytics {
...
+      loadOptions = dataPlaneUrl;
```
[the remark this answers is not in the document]
Comment (bardisg) [reply packages/analytics-js/src/components/core/Analytics.ts:143]: just assign to a new variable and use the new one to update state

File: packages/analytics-js/src/components/configManager/ConfigManager.ts [hunk 3]
```diff
@@ -125,7 +125,7 @@ class ConfigManager implements IConfigManager {
   processConfig(response?: SourceConfigResponse | string, rejectionDetails?: RejectionDetails) {
     // TODO: add retry logic with backoff based on rejectionDetails.hxr.status
     if (!response) {
-      this.onError(`Unable to fetch source config ${rejectionDetails?.error}`, undefined, true);
+      this.onError(`Unable to fetch source config ${rejectionDetails?.error}`);
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply packages/analytics-js/src/components/configManager/ConfigManager.ts:128]: We wanted to stop the execution then and there. As we are returning the execution after that line so it is fine.

**Comment by MoumitaM** [comment]
All comments are addressed @bardisg and @saikumarrs

**Pull request merged**